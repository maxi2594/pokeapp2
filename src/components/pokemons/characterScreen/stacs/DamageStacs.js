import React from 'react'
import PropTypes from 'prop-types';

import { ExelColums } from '../../../commons/ExelColums';

export const DamageStacs = ({ damage }) => {

    const damageArr = Object.entries(damage)

    return (
        <div className='typeStacs'>
            <div className='statsTitle'>
                <h4>Damage Stacs</h4>
            </div>
            {
                damageArr.map(elem =>
                (<ExelColums
                    key={elem[0]}
                    item={elem[0]}
                    values={elem[1]}
                />)
                )
            }

        </div>
    )
}
DamageStacs.propTypes = {
    damage: PropTypes.object.isRequired
}