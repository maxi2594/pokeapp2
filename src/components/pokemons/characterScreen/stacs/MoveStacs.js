import React, { useState } from 'react'
import PropTypes from "prop-types";

import { ExelColums } from '../../../commons/ExelColums';
import { DescriptionBox } from '../../../commons/DescriptionBox';

export const MoveStacs = ({
    move_damage_class: classMove,
    moves,
}) => {

    const [activeMove, setactiveMove] = useState({
        name: moves[0].name,
        url: moves[0].url
    })

    const handleClick = (elem) => {
        setactiveMove(s => ({
            ...s,
            name: elem.name,
            url: elem.url
        }))
    }
    return (
        <div className='typeStacs'>

            <div className='statsTitle'>
                <h4>Moves</h4>
            </div>

            <ExelColums
                key={classMove.name}
                item={'Damage Class'}
                values={[classMove]}
            />
            <div className='movesBox'>
                {
                    moves.map(elem =>
                    (<button
                        // className={(typeStacs.type === elem.type.name) ? 'active' : ''}
                        key={elem.name}
                        onClick={() => handleClick(elem)}
                    >{elem.name}</button>)
                    )
                }
            </div>

            <DescriptionBox data={activeMove} />
        </div>
    )
}

MoveStacs.propTypes = {
    move_damage_class: PropTypes.object.isRequired,
    moves: PropTypes.array.isRequired
}
