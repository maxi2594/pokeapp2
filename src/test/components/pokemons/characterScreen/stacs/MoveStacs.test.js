import { render, screen } from "@testing-library/react";
import { shallow } from "enzyme"

import { MoveStacs } from "../../../../../components/pokemons/characterScreen/stacs/MoveStacs"
import { data } from "../../../../fakeData/fakeData"


describe('test in <MoveStacs />', () => {

    test('should render Properly', () => {

        const wrapper = shallow(<MoveStacs {...data} />)

        expect(wrapper).toMatchSnapshot()
    })

    test('should appear data in screen', () => {

        const { getAllByRole, getByText } = render(<MoveStacs {...data} />)

        expect(getByText(
            new RegExp(data.move_damage_class.name, 'i')
        )).toBeInTheDocument()

        const buttons = getAllByRole('button');

        expect(buttons.length).toEqual(data.moves.length)
    })


})