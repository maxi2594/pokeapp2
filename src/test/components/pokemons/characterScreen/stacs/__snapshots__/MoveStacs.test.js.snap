// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in <MoveStacs /> should render Properly 1`] = `
<div
  className="typeStacs"
>
  <div
    className="statsTitle"
  >
    <h4>
      Moves
    </h4>
  </div>
  <ExelColums
    item="Damage Class"
    key="special"
    values={
      Array [
        Object {
          "name": "special",
          "url": "https://pokeapi.co/api/v2/move-damage-class/3/",
        },
      ]
    }
  />
  <div
    className="movesBox"
  >
    <button
      key="rock-throw"
      onClick={[Function]}
    >
      rock-throw
    </button>
    <button
      key="rock-slide"
      onClick={[Function]}
    >
      rock-slide
    </button>
    <button
      key="sandstorm"
      onClick={[Function]}
    >
      sandstorm
    </button>
  </div>
  <DescriptionBox
    data={
      Object {
        "name": "rock-throw",
        "url": "https://pokeapi.co/api/v2/move/88/",
      }
    }
  />
</div>
`;
